### DISCUSSION TITLE
Continuing the work

### DISCUSSION THREAD

**Discussion by LCMilstein** [pypolestar/polestar_api#147] [General]
I'm so grateful this integration exists. I would love to contribute, but admit I don't yet have the technical skills to be of much use to this community. Is anyone else picking up this repo or creating a fork anywhere to add functions? The main things I'd love to help support are status and control of the door locks, windows, and trunk. Anyone see this working anywhere?

**Comment by leeyuentuen** [comment]
same as this issue: 
https://github.com/leeyuentuen/polestar_api/issues/145

but i can still maintain it or if someone can shared there credential to continue fix the issue or new feature

on the other side, the extra feature like control is a duplicate question. already mention before that this is a limitation of the current api as mention here 
https://github.com/leeyuentuen/polestar_api/discussions/114

**Reply by LCMilstein** [reply]
Amazing. Thanks. And I saw the discussion about the api limitations such that control might not be possible, but I think status of doors, locks, windows and trunk should be doable. If there's no better way, those statuses are reported via the Google Assistant API if enabled in assistant settings. Since there's an existing Google Assistant integration so that these status should be queryable from Home Assistant with that integration enabled, so it should be possible to store the results in ways that can be reported on a dash...

**Reply by olalid** [reply]
It is a common misconception that you can query polestar vehicle status through home assistant google integration, but in fact google has put an extra security level on vehicles specifically which third party integrations can not access. Only actual voice queries are possible.

**Reply by LCMilstein** [reply]
The documentation says, the integration enables 

> > "Sending text commands to Google Assistant to control devices supported by Google Assistant but not by Home Assistant. Examples:
> 
> - Set Nest Guard to home and guarding
> - Start vacuuming
> - Stream front door on living room TV
> - Turn off kitchen TV
> - Play rain sounds on bedroom speaker

Since I can query Google Assistant ("Are my windows up") and the Assistant can reply, why wouldn't it be possible to capture that reply, use Home Assistant voice to parse that reply, or use an LLM to parse that reply, and store it in a Helper?

**Reply by BeauGiles** [reply]
> The documentation says, the integration enables
> 
> > > "Sending text commands to Google Assistant to control devices supported by Google Assistant but not by Home Assistant. Examples:
> > 
> > 
> > 
> > * Set Nest Guard to home and guarding
> > * Start vacuuming
> > * Stream front door on living room TV
> > * Turn off kitchen TV
> > * Play rain sounds on bedroom speaker
> 
> Since I can query Google Assistant ("Are my windows up") and the Assistant can reply, why wouldn't it be possible to capture that reply, use Home Assistant voice to parse that reply, or use an LLM to parse that reply, and store it in a Helper?

I think because it needs to verify your identity, so it won't work.

> Examples of known Google Assistant API issues:
Media playback commands (other than play news, play podcast, play white noise, or play rain sounds) don’t work.
Routines don’t work.
Broadcast to specific rooms often doesn’t work for non-English languages.
Commands that need to verify your identity through voice match do not work.
https://www.home-assistant.io/integrations/google_assistant_sdk/#limitationsknown-issues

**Reply by garyellis1** [reply]
> but i can still maintain it or if someone can shared there credential to continue fix the issue or new feature

I'm happy to share if it helps.

**Comment by wouterhardeman** [comment]
It's not possible due to requiring Voice Match which is only available on approved applications and devices.

![Screenshot from 2024-06-19 00-14-31](https://github.com/leeyuentuen/polestar_api/assets/495620/51b9f497-30fe-41fb-a4de-1953e4e9deb4)

The top query is done through Home Assistant via the Google Assistant SDK and the bottom one is through the Google Assistant app on my phone.

**Reply by LCMilstein** [reply]
hmm... Interesting. I guess we'll need other ideas... 
In case anyone isn't aware, I was able to get the device_tracker.polestar entity working by enabling the permission via the in-car Home Assistant app settings.
